### PR TITLE
Formatting code by running prettier

### PR DESCRIPTION
**Pull request by johnsimons**
This is so we can standardise on prettier.

I tried a few options on `printWidth`, but I settled for 250 for now, given the smaller number of files changed.

I also had to update a few packages (eslint + prettier) so that things worked properly.

### CODE REVIEW

File: src/ServicePulse.Host/vue/package.json [hunk 1]
```diff
@@ -24,15 +25,15 @@
     "vue3-simple-typeahead": "^1.0.11"
   },
   "devDependencies": {
-    "@rushstack/eslint-patch": "^1.1.4",
+    "@rushstack/eslint-patch": "^1.7.0",
```
[the remark this answers is not in the document]
Comment (johnsimons) [reply src/ServicePulse.Host/vue/package.json:28]: Just to ensure there are no surprises once we update.
I also found that the latest picked up a few more issues

File: src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue [hunk 2]
```diff
@@ -12,6 +12,7 @@ const settings = defineProps({
 });
 
 let panel = ref();
+// eslint-disable-next-line vue/no-dupe-keys
```
[the remark this answers is not in the document]
Comment (johnsimons) [reply src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue:15]: This is me not knowing enough vue, I'll try to figure it out how to fix it

File: src/ServicePulse.Host/vue/package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
     "build": "npm run lint && vite build",
     "preview": "npm run lint && vite preview",
     "test:unit": "npm run lint && vitest --environment jsdom --root src/",
-    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs --fix"
+    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs --fix",
+    "format": "prettier --write src/"
```
[the remark this answers is not in the document]
Comment (johnsimons) [reply src/ServicePulse.Host/vue/package.json:12]: I'll remove it

[the remark this answers is not in the document]
Comment (johnsimons) [reply src/ServicePulse.Host/vue/package.json]: @PhilBastian i have added the [start of docs](https://github.com/Particular/ServicePulse/pull/1673/commits/bcb3559667df76d64fd9dd82e0151865fc5b2d4b) regarding IDE, but I am familiar with Webstorm.
And I think given you are more familiar with VSCode, can you add extra info on it?

**Comment by PhilBastian** [comment]
need to add Prettier as a requirement to dev setup in the readme

**Comment by mikeminutillo** [comment]
> need to add Prettier as a requirement to dev setup in the readme

It can also be added to the `.vscode\extensions.json` so that people get prompted to install it. I think that is excluded via `.gitignore`

**Comment by PhilBastian** [comment]
> > need to add Prettier as a requirement to dev setup in the readme
> 
> It can also be added to the `.vscode\extensions.json` so that people get prompted to install it. I think that is excluded via `.gitignore`

I deleted the original comment because I realised John had since added the missing readme changes. The issue with the extensions.json is that it will only be for VSCode and also will install the extension without configuring it to format on save

**Review by PhilBastian — APPROVED** [review]
Formatting changes are approved, needs further work by the TF on documentation changes

File: docs/frontend/frontend-ide.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+#### Keymap
+
+You can change your KeyMap under `File | Settings | Keymap`. A good option is the `Visual Studio` keymap, as this will give you consistent keyboard shortcuts with Visual Studio / Rider, and is also the most commonly used keymap by developers at Particular.
+
+#### Other settings
+
+- To improve the quality of suggestions you receive when using code completion, go to `File | Settings | Editor | General | Code Completion` and enable `Only type-based completion`.
+
+## VSCode
```
Comment (mikeminutillo) [inline docs/frontend/frontend-ide.md:32]: Should this include Prettier and the Vue.volar extensions?

File: src/ServicePulse.Host/vue/src/components/failedmessages/MessageGroupList.vue [hunk 5]
```diff
@@ -412,22 +449,38 @@ defineExpose({
...
+                                <i aria-hidden="true" class="fa fa-exclamation-triangle danger"></i> <strong>WARNING: </strong>Not all messages will be retried because ServiceControl had to restart. You need to request retrying the remaining
+                                messages.
+                              </div>
                             </li>
                           </ul>
 
                           <div class="op-metadata">
-                            <span class="metadata"><i aria-hidden="true" class="fa fa-envelope"></i> {{ group.workflow_state.status === "completed" ? "Messages sent:" : "Messages to send:" }} {{ group.operation_remaining_count || group.count }}</span>
+                            <span class="metadata"
+                              ><i aria-hidden="true" class="fa fa-envelope"></i> {{ group.workflow_state.status === "completed" ? "Messages sent:" : "Messages to send:" }} {{ group.operation_remaining_count || group.count }}</span
```
Comment (mikeminutillo) [inline src/ServicePulse.Host/vue/src/components/failedmessages/MessageGroupList.vue:478]: These (closing bracket on newline) look weird. Is this due to a significant whitespace thing inside of spans?

**Review by mikeminutillo — APPROVED** [review]

File: docs/frontend/frontend-ide.md (hunk 4 above)
Comment (PhilBastian) [reply docs/frontend/frontend-ide.md:30]: Vue.volar is suggested by vscode itself, and not having it won't affect others like not having prettier/eslint will

File: src/ServicePulse.Host/vue/src/components/failedmessages/MessageGroupList.vue (hunk 5 above)
Comment (PhilBastian) [reply src/ServicePulse.Host/vue/src/components/failedmessages/MessageGroupList.vue:478]: appears to be prettier freaking out in handling this particular combination. Fixed by forcing our own newlines

**Pull request merged**